src/worker.js

```diff
@@ -239,6 +239,9 @@ this.onmessage = function(e) {
 #if WASM_BACKEND
       Module['_emscripten_tls_init']();
 #endif
+#if SAFE_STACK
+      Module['___set_stack_limit'](STACK_MAX);
+#endif
```

Comment:
It would be good to add a test that this works with pthreads as well, since we we add support for it here.

Reply:
Test added.